src/app/view/components/dot-navigation/dot-navigation.component.ts

```diff
@@ -12,13 +12,30 @@ import { DotNavigationService } from './dot-navigation.service';
 })
 export class DotNavigationComponent implements OnInit {
     menu: Observable<DotMenu[]>;
+    @Output() onMenuLinkClicked = new EventEmitter<Event>();
 
-    constructor(private dotNavigationService: DotNavigationService) {}
+    constructor(private dotNavigationService: DotNavigationService, private el: ElementRef) {}
 
     ngOnInit() {
         this.menu = this.dotNavigationService.items$;
     }
 
+    /**
+     * Respond to document events and collapse the sidenav if is clicked outside
+     * TODO: not working on iframes
+     * @param {*} event
+     * @memberof DotNavigationComponent
+     */
+    @HostListener('document:click', ['$event'])
+    handleClick(event: any) {
+        if (!this.el.nativeElement.contains(event.target)) {
```

Comment:
This condition it's too complicated, double check it and move this to a method with a name that tells what is this condition about.